src/auth/oauth2client.ts

```diff
@@ -705,6 +721,21 @@ export class OAuth2Client extends AuthClient {
     return {headers, res: r.res};
   }
 
+  /**
+   * Generates an URL to revoke the given token.
+   * @param token The existing token to be revoked.
+   * @param jsonp If set, append callback= parameter to request JSONP response.
+   */
+  static getRevokeTokenUrl(token: string, jsonp?: string): string {
```

Comment:
Does this really need to be public?

Reply:
I would better make it public to make it possible to assign the URL to some button, etc. Can be discussed though.

Reply:
Well, it must be public. The server does not send CORS headers for revoking the token, so the only way to call it from browser is via JSONP (by adding a `<script>` tag that will point to that URL), so the URL must be exposed by the library.
![image](https://user-images.githubusercontent.com/4015807/40881220-21b3cb8e-6676-11e8-87f0-b9470772392d.png)

Other calls (such as getting a token) return CORS headers so we don't need to expose their URLs. If I find more URLs without CORS I'll need to expose them as well.

Reply:
An alternative is to make a function that will append `<script>` tag and make all JSONP magic happen under the hood (like JQuery's `$.ajax({ jsonp: functionName })` call does), but I don't like this idea since I feel it's very wrong for this auth library to make any assumptions about HTML document structure and spam with extra `<script>` tags.